packages/server/src/distributed/local/subservice.ts

```diff
@@ -1,11 +1,27 @@
 import { DistributedSubservice } from '../base/subservice'
+import { Lock } from '../types'
 
 export class LocalSubservice implements DistributedSubservice {
+  private locks: { [ressource: string]: boolean } = {}
+
   async setup() {}
 
   async destroy() {}
 
   async listen(channel: string, callback: (message: any) => Promise<void>) {}
 
   async send(channel: string, message: any) {}
+
+  async lock(ressource: string): Promise<Lock> {
+    if (!this.locks[ressource]) {
+      this.locks[ressource] = true
+      return { ressource }
+    } else {
+      throw new Error('Failed to acquire lock. Already in use.')
```

Comment:
Does it mean that if you are running a single instance and try to acquire the same lock to different places, it will throw an error instead of waiting for it to be released? If so, is this what we want? 

Also, what if release is never called?

Reply:
If release is not called the lock will time out after 2 seconds. We don't need to explicitly call release. The expiry is stored on redis so other nodes can know when a lock is timed out.

We want to throw an error is some circumstances. In other we might want to skip over the bit of code if the lock exists. It depends.

Reply:
> If release is not called the lock will time out after 2 seconds. We don't need to explicitly call release. The expiry is stored on redis so other nodes can know when a lock is timed out.

I understand, but what if you are not using Redis?

> We want to throw an error is some circumstances. In other we might want to skip over the bit of code if the lock exists. It depends.

Fair enough!

Reply:
Good point it doesn't work with the local locking. I'll put an expiry time in there instead of a boolean